### PR TITLE
Move remaining ownership emails from the Controller layer to the service layer

### PR DESCRIPTION
**Pull request by joelverhagen**
Move the rest of ownership related emails into the service layer so an admin panel can send the mails properly as well without duplicating code.

Depends on https://github.com/NuGet/NuGetGallery/pull/8769.
Progress on https://github.com/NuGet/Engineering/issues/4034.
Related to https://github.com/NuGet/Engineering/issues/3994.

### CODE REVIEW

File: src/NuGetGallery.Services/PackageManagement/IPackageOwnershipManagementService.cs [hunk 1]
```diff
@@ -36,6 +46,13 @@ public interface IPackageOwnershipManagementService
         /// <param name="newOwner">The user to be added for from pending ownership.</param>
         Task<PackageOwnerRequest> AddPackageOwnershipRequestAsync(PackageRegistration packageRegistration, User requestingOwner, User newOwner);
 
+        /// <summary>
+        /// Remove the user as from the list of owners of the package and then sends notification messages. Immediately
+        /// commits the changes to the database. Same behavior as <see cref="RemovePackageOwnerAsync(PackageRegistration, User, User, bool)" />
+        /// with the addition of sending messages.
+        /// </summary>
```
Comment (zhhyu) [inline src/NuGetGallery.Services/PackageManagement/IPackageOwnershipManagementService.cs:53]: Parameter comments?

File: src/NuGetGallery.Services/PackageManagement/PackageOwnershipManagementService.cs [hunk 2]
```diff
@@ -132,6 +133,69 @@ private async Task AddPackageOwnerTask(PackageRegistration packageRegistration,
             await DeletePackageOwnershipRequestAsync(packageRegistration, user, commitChanges, saveAudit: false);
         }
 
+        public async Task<PackageOwnerRequest> AddPackageOwnershipRequestWithMessagesAsync(
+            PackageRegistration packageRegistration,
+            User requestingOwner,
+            User newOwner,
+            string message)
+        {
```
Comment (zhhyu) [inline src/NuGetGallery.Services/PackageManagement/PackageOwnershipManagementService.cs:141]: Should we also have some null checks here? I can find it at Remove, Cancel and Decline.

**Review by zhhyu — APPROVED** [review]

File: src/NuGetGallery.Services/PackageManagement/PackageOwnershipManagementService.cs [hunk 3]
```diff
@@ -11,6 +11,7 @@
 using NuGet.Services.Messaging.Email;
 using NuGetGallery.Infrastructure.Mail.Messages;
 using NuGetGallery.Configuration;
+using System.Web;
```
Comment (loic-sharma) [inline src/NuGetGallery.Services/PackageManagement/PackageOwnershipManagementService.cs:14]: Super nit: this is out of order

File: tests/NuGetGallery.Facts/Services/PackageOwnershipManagementServiceFacts.cs [hunk 4]
```diff
@@ -316,8 +316,101 @@ public async Task WritesAnAuditRecord()
...
+                        It.IsAny<bool>()));
+                messageService.Verify(
+                    x => x.SendMessageAsync(
+                        It.Is<PackageOwnershipRequestMessage>(m => m.ToUser == newOwner && m.FromUser == requestingOwner),
+                        It.IsAny<bool>(),
+                        It.IsAny<bool>()));
+            }
+
+            [Fact]
+            public async Task SendsAllMessagesEvenIfOneFails()
```
Comment (loic-sharma) [inline tests/NuGetGallery.Facts/Services/PackageOwnershipManagementServiceFacts.cs:356]: FYI this is not 100% guaranteed and is dependent on the implementation of the `SendMessageAsync` method. See the exception note here: https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md#prefer-asyncawait-over-directly-returning-task

I'm just being pedantic, I don't think we should try to solve this problem :)

**Review by loic-sharma — APPROVED** [review]

File: src/NuGetGallery.Services/PackageManagement/IPackageOwnershipManagementService.cs (hunk 1 above)
Comment (joelverhagen) [reply src/NuGetGallery.Services/PackageManagement/IPackageOwnershipManagementService.cs:53]: Nice catch!

File: src/NuGetGallery.Services/PackageManagement/PackageOwnershipManagementService.cs (hunk 2 above)
Comment (joelverhagen) [reply src/NuGetGallery.Services/PackageManagement/PackageOwnershipManagementService.cs:141]: I don't think we have well-defined standards. But we do have a standard of being consistent so I'll invoke that rule here 😄. Good catch.

File: tests/NuGetGallery.Facts/Services/PackageOwnershipManagementServiceFacts.cs (hunk 4 above)
Comment (joelverhagen) [reply tests/NuGetGallery.Facts/Services/PackageOwnershipManagementServiceFacts.cs:356]: Ah great point. I'll leave the implementation as-is due to the known implementation of these methods.

**Pull request merged**